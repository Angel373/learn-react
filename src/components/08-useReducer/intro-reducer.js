const initialState = [
    {
        id: 1,
        todo: 'Comprar pan',
        done: false
    }
]

const todoReduce = ( state = initialState, action) => {

    if (action?.type === 'agregar') {
       return [...state, action.payload]; 
    }
    return state;

}

let todos = todoReduce();

const newTodo = {
    id: 2,
    todo: 'comprar leche',
    done: false
}

const action = {
    type: 'agregar',
    payload: newTodo
}
todos = todoReduce(todos, action);
console.log(todos);
