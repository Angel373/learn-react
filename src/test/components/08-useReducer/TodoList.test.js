import { shallow } from "enzyme"
import { TodoList } from "../../../components/08-useReducer/TodoList"
import React from 'react'
import { demoTodos } from "../fixtures/demoTodos"

describe('pruebas en TodoList', () => {

    const handleToggle = jest.fn();
    const handleDelete = jest.fn();

    const wrapper = shallow(<TodoList
        todos = {demoTodos}
        handleToggle = {handleToggle}
        handleDelete = {handleDelete}
    />);
    test('debe de mostrar correctamente el componente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('debe de tener dos <TodoListItem />', () => {
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));
    })
    
    
})
