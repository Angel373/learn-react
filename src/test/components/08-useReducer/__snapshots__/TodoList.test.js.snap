// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en TodoList debe de mostrar correctamente el componente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    todo={
      Object {
        "desc": "Aprender react",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    todo={
      Object {
        "desc": "Aprender mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
